controlplane/api/v1beta2/ck8scontrolplane_webhook.go

```diff
@@ -76,4 +77,29 @@ func defaultCK8sControlPlaneSpec(s *CK8sControlPlaneSpec, namespace string) {
 	if s.MachineTemplate.InfrastructureRef.Namespace == "" {
 		s.MachineTemplate.InfrastructureRef.Namespace = namespace
 	}
+
+	s.RolloutStrategy = defaultRolloutStrategy(s.RolloutStrategy)
+}
+
+func defaultRolloutStrategy(rolloutStrategy *RolloutStrategy) *RolloutStrategy {
+	ios1 := intstr.FromInt(1)
+
+	if rolloutStrategy == nil {
+		rolloutStrategy = &RolloutStrategy{}
+	}
+
+	// Enforce RollingUpdate strategy and default MaxSurge if not set.
+	if rolloutStrategy != nil {
+		if len(rolloutStrategy.Type) == 0 {
+			rolloutStrategy.Type = RollingUpdateStrategyType
+		}
+		if rolloutStrategy.Type == RollingUpdateStrategyType {
+			if rolloutStrategy.RollingUpdate == nil {
+				rolloutStrategy.RollingUpdate = &RollingUpdate{}
+			}
+			rolloutStrategy.RollingUpdate.MaxSurge = intstr.ValueOrDefault(rolloutStrategy.RollingUpdate.MaxSurge, ios1)
```

Comment:
question: is the given `rolloutStrategy.Type` verified anywhere? What if I have typo? Same with `MaxSurge` - we say we support the values of 0 and 1, but what happens if we're given `MaxSurge` -2, or 5?

Reply:
ditto from below.